### ISSUE TITLE
bug: Cannot send to SQS queue using queue name, queue url is required after upgrading to 2.1.0

### ISSUE THREAD

**Issue by lbenedetto** [localstack/localstack#8545]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have code which sends a message to an SQS queue:

```java
amazonSQS.sendMessage(new SendMessageRequest()
    .withMessageBody(messageBody)
    .withMessageAttributes(messageAttributes))
    .withQueueUrl(queueName));
```
This code works in production and in localstack 2.0.2

However, after updating localstack, it throws an exception:
```
Caused by: com.amazonaws.services.sqs.model.QueueDoesNotExistException: The specified queue does not exist for this wsdl version. (Service: AmazonSQS; Status Code: 400; Error Code: AWS.SimpleQueueService.NonExistentQueue; Request ID: 7a5b02f7-167d-4cb5-9cea-618062064e4b; Proxy: null)
```

The error goes away if I send a queueUrl (like `http://localstack:4566/000000000000/MyMessage`) instead of a queueName (like `MyMessage`)
```java
.withQueueUrl(sqsClient.getQueueUrl(queueName).getQueueUrl())
// or
.withQueueUrl("000000000000/" + queueName)
```

I have captured some logs from the error (and manually formatted the request and response bodies for readability):
Localstack 2.0.2
```ruby
# 2023-06-21T22:00:04.256 DEBUG --- [ asgi_gw_751] l.aws.serving.wsgi         : POST localhost:14005/MyMessage
# 2023-06-21T22:00:04.259 DEBUG --- [ asgi_gw_751] l.aws.protocol.serializer  : No accept header given. Using request's Content-Type (application/x-www-form-urlencoded; charset=utf-8) as preferred response Content-Type.
# 2023-06-21T22:00:04.259  INFO --- [ asgi_gw_751] localstack.request.aws     : AWS sqs.SendMessage => 200;

SendMessageRequest(
{
    'QueueUrl': None, 
    'MessageBody': '{...}', 
    'MessageAttributes': {...}
}, 
headers={
    'Host': 'localhost:14005',
    'amz-sdk-invocation-id': '34a2dfd8-476f-0b7b-bcdf-b8dabff03a12', 
    'amz-sdk-request': 'ttl=20230621T220054Z;attempt=1;max=4', 
    'amz-sdk-retry': '0/0/500', 
    'Authorization': 'AWS4-HMAC-SHA256 Credential=test/20230621/eu-west-1/sqs/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;amz-sdk-retry;host;user-agent;x-amz-date, Signature=f79b5623feaa22faf1f8a499eb4c708c3f390a3458c1eaa874acda2a427b56d0', 
    'User-Agent': 'aws-sdk-java/1.12.488 Linux/5.15.0-75-generic OpenJDK_64-Bit_Server_VM/17.0.6+10-LTS java/17.0.6 kotlin/1.8.21 vendor/Amazon.com_Inc. cfg/retry-mode/legacy AmazonSQSExtendedClient/1.12.488', 
    'X-Amz-Date': '20230621T220004Z', 
    'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 
    'Content-Length': '846', 
    'Connection': 'Keep-Alive', 
    'x-localstack-tgt-api': 'sqs', 
    'x-moto-account-id': '000000000000'
});

SendMessageResult(
{
    'MessageId': 'f363c73c-56f1-412a-980d-3717b5aa05de', 
    'MD5OfMessageBody': '974c75dc0c2d71128fd4e1e6283d1141',
    'MD5OfMessageAttributes': '1910dd2d542ced919f469c29378d01cb', 
    'SequenceNumber': None, 
    'MD5OfMessageSystemAttributes': None
 }, 
 headers={
    'Content-Type': 'text/xml', 
    'Content-Length': '493',
    'Access-Control-Allow-Origin': '*', 
    'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH',
    'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request',
    'Access-Control-Expose-Headers': 'etag,x-amz-version-id'
})
```
Localstack 2.1.0

```ruby
# 2023-06-21T22:01:54.669 DEBUG --- [   asgi_gw_1] l.aws.serving.wsgi         : POST localhost:14005/MyMessage
# 2023-06-21T22:01:54.671 DEBUG --- [   asgi_gw_1] l.aws.protocol.serializer  : No accept header given. Using request's Content-Type (application/x-www-form-urlencoded; charset=utf-8) as preferred response Content-Type.
# 2023-06-21T22:01:54.671  INFO --- [   asgi_gw_1] localstack.request.aws     : AWS sqs.SendMessage => 400 (AWS.SimpleQueueService.NonExistentQueue);

SendMessageRequest(
{
    'QueueUrl': None, 
    'MessageBody': '{...}', 
    'MessageAttributes': {...}
}, 
headers={
    'Host': 'localhost:14005', 
    'amz-sdk-invocation-id': '66ada625-ed81-5b82-ef87-10e5b74c3081',
     'amz-sdk-request': 'ttl=20230621T220244Z;attempt=1;max=4', 
     'amz-sdk-retry': '0/0/500', 
     'Authorization': 'AWS4-HMAC-SHA256 Credential=test/20230621/eu-west-1/sqs/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;amz-sdk-retry;host;user-agent;x-amz-date, Signature=9b1776f8d73372c548828f40d27b517e3a02a1234662a9d348108952a0af2c41', 
     'User-Agent': 'aws-sdk-java/1.12.488 Linux/5.15.0-75-generic OpenJDK_64-Bit_Server_VM/17.0.6+10-LTS java/17.0.6 kotlin/1.8.21 vendor/Amazon.com_Inc. cfg/retry-mode/legacy AmazonSQSExtendedClient/1.12.488', 
     'X-Amz-Date': '20230621T220154Z', 
     'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 
     'Content-Length': '844', 
     'Connection': 'Keep-Alive',
     'x-localstack-tgt-api': 'sqs', 
     'x-moto-account-id': '000000000000'
 }); 
AWS.SimpleQueueService.NonExistentQueue(The specified queue does not exist for this wsdl version., headers={
    'Content-Type': 'text/xml',
    'Content-Length': '342', 
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 
    'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 
    'Access-Control-Expose-Headers': 'etag,x-amz-version-id'
})
```

### Expected Behavior

localstack 2.x should work with both queueName and queueUrl, since that is the behavior of actual SQS.

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
```yml
  localstack:
    image: registry.ops-euw1.tgtg.ninja/localstack/localstack:2.1.0
    environment:
      - SERVICES=sns,sqs,cloudformation,s3
      - HOSTNAME=localstack
      - HOSTNAME_EXTERNAL=${LOCALSTACK_EXTERNAL}
      - DEFAULT_REGION=eu-west-1
      - AWS_ACCESS_KEY_ID=test
      - AWS_SECRET_ACCESS_KEY=test
      - SQS_PROVIDER=elasticmq
      - TMPDIR=/tmp
      - DEBUG=1
      - LS_LOG=trace
      - HOST_TMP_FOLDER=${CURRENT_DIR}/.cache/localstack/tmp
    volumes:
      - ${CURRENT_DIR}/.cache/localstack:/usr/stuff
      - ${CURRENT_DIR}/.cache/localstack/tmp:/var/lib/localstack
      - ${CURRENT_DIR}/docker/localstack/s3:/usr/stuff/s3
      - ${CURRENT_DIR}/docker/localstack/init.sh:/etc/localstack/init/ready.d/init.sh
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - "14005:4566"
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Unfortunately, I was unable to reproduce the issue from the command line as all versions of localstack I tested exhibited identical behavior for these 4 commands:
```bash
aws sqs send-message --queue-url MyMessage --message-body asdf --endpoint-url=http://localstack:4566 --region eu-west-1
aws sqs send-message --queue-url MyMessage --message-body asdf --region eu-west-1
aws sqs send-message --queue-url http://localstack:4566/000000000000/MyMessage --message-body asdf --endpoint-url=http://localstack:4566 --region eu-west-1
aws sqs send-message --queue-url http://localstack:4566/000000000000/MyMessage --message-body asdf --region eu-west-1
```

### Environment

```markdown
- OS:Linux Mint 21.1
- LocalStack: 2.1.0
```

### Anything else?

Looking through the changelog for 2.1.0 it seems to me that the most likely culprit is this PR: [fix account resolution in SQS QueryAPI operations ](https://github.com/localstack/localstack/pull/8353), but I haven't looked at it too closely, I'm just basing that on the title.

**Comment by thrau** [comment]
Thanks for reporting Lars @lbenedetto. I also wasn't able to reproduce the issue with the latest version and the aws CLI. 

The Java SDK has been known to be a bit peculiar with older services like SQS and S3, so we'll have to take a closer look with a Java project.

On a sidenote, it seems your docker-compose configuration uses several deprecated/unused variables. Here's a cleaned up version:

```yaml
  localstack:
    image: registry.ops-euw1.tgtg.ninja/localstack/localstack:2.1.0
    environment:
      - LOCALSTACK_HOST=localstack
      - DEFAULT_REGION=eu-west-1
      - AWS_ACCESS_KEY_ID=test
      - AWS_SECRET_ACCESS_KEY=test
      - DEBUG=1
      - LS_LOG=trace
    volumes:
      - ${CURRENT_DIR}/.cache/localstack:/usr/stuff
      - ${CURRENT_DIR}/.cache/localstack/tmp:/var/lib/localstack
      - ${CURRENT_DIR}/docker/localstack/s3:/usr/stuff/s3
      - ${CURRENT_DIR}/docker/localstack/init.sh:/etc/localstack/init/ready.d/init.sh
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - "14005:4566"
```

**Comment by lbenedetto** [comment]
Here is a java project which reproduces the issue.

https://github.com/lbenedetto/LocalstackIssue8545Reproducer

First run `docker-compose up`. This will start two instances of localstack, one running 2.0.2 (on port 14202) and one running 2.1.0 (on port 14210).

Then run `./gradlew test`

This will run 4 tests:
 * testListQueuesWithLocalstack202 -> Verifies that queue exists. Passing
 * testListQueuesWithLocalstack210 -> Verifies that queue exists. Passing
 * testSendMessageWithLocalstack202 -> Verifies that sending message works. Passing
 * testSendMessageWithLocalstack210 -> Verifies that sending message works. Failing :(

**Comment by lbenedetto** [comment]
This is resolved in localstack 3.0.0
